great_expectations_provider/example_dags/example_great_expectations_dag.py

```diff
@@ -0,0 +1,164 @@
+    can load our CSV files from the data directory."""
+    data_source = context.data_sources.add_pandas_filesystem(
+        name="Extract Data Source",
+        base_directory=data_dir,
+    )
+    asset = data_source.add_csv_asset(name="Extract CSV Asset")
+    batch_definition = asset.add_batch_definition_monthly(
+        name="Extract Batch Definition",
+        regex="yellow_tripdata_sample_(?P<year>\d{4})-(?P<month>\d{2}).csv",
+    )
+    return batch_definition
+
+
+def configure_checkpoint(context: AbstractDataContext) -> Checkpoint:
+    """This function takes a GX Context and returns a Checkpoint that
+    can load our CSV files from the data directory, validate them
+    against an ExpectationSuite, and run Actions."""
+    # setup data source, asset, batch definition
+    batch_definition = (
+        context.data_sources.add_pandas_filesystem(
+            name="Load Datasource", base_directory=data_dir
+        )
+        .add_csv_asset("Load Asset")
+        .add_batch_definition_monthly(
+            name="Load Batch Definition",
+            regex="yellow_tripdata_sample_(?P<year>\d{4})-(?P<month>\d{2}).csv",
+        )
+    )
+    # setup expectation suite
+    expectation_suite = context.suites.add(
+        ExpectationSuite(
+            name="Load ExpectationSuite",
+            expectations=[
+                gxe.ExpectTableRowCountToBeBetween(
+                    min_value=9000,
+                    max_value=11000,
+                ),
+                gxe.ExpectColumnValuesToNotBeNull(column="vendor_id"),
+                gxe.ExpectColumnValuesToBeBetween(
+                    column="passenger_count", min_value=1, max_value=6
+                ),
+            ],
+        )
+    )
+    # setup validation definition
+    validation_definition = context.validation_definitions.add(
+        ValidationDefinition(
+            name="Load Validation Definition",
+            data=batch_definition,
+            suite=expectation_suite,
+        )
+    )
+    # setup checkpoint
+    checkpoint = context.checkpoints.add(
+        Checkpoint(
+            name="Load Checkpoint",
+            validation_definitions=[validation_definition],
+            actions=[],
+        )
+    )
+    return checkpoint
+
+
+# Batch Parameters will also be available as DAG params, to be consumed directly by the
+# operator through the context. Users can still provide batch_parameters on operator init
+# (critical for validating data frames), but batch_parameters provided as DAG params should take precedence.
+# To demo validation failure, use FAILURE_MONTH as a batch parameter instead of SUCCESS_MONTH
+SUCCESS_MONTH = "01"
+FAILURE_MONTH = "02"
+batch_parameters = {"year": "2019", "month": SUCCESS_MONTH}
+
+
+with DAG(
+    dag_id="gx_provider_example_dag",
+) as dag:
+    # define a consistent set of expectations we'll use throughout the pipeline
+    expectation_suite = ExpectationSuite(
```

Comment:
Since this is not an Airflow task can we move this ExpectationSuite outside of the dag definition?

Reply:
sure, i'll move it out of the DAG, thanks